### PR TITLE
fix: 'window count is over max!!' 

### LINKED ISSUE #15658: [Bug] IllegalStateException: window count is over max!!
https://ankidroid.org/acra/app/1/bug/132331/report/268484d3-3378-44a2-a3c5-59ceac0d7a09

```
java.lang.IllegalStateException: window count is over max!!
	at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:450)
	at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:148)
	at android.app.Dialog.show(Dialog.java:521)
	at androidx.appcompat.app.AlertDialog$Builder.show(SourceFile:5)
	at com.ichi2.anki.pages.PageChromeClient.onJsAlert(SourceFile:51)
	at WV.z6.run(chromium-TrichromeWebViewGoogle6432.aab-stable-616717833:138)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8762)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:604)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

Note:

```
	at androidx.appcompat.app.AlertDialog$Builder.show(SourceFile:5)
	at com.ichi2.anki.pages.PageChromeClient.onJsAlert(SourceFile:51)
```

Seen a couple of times in ACRA. Needs reproduction steps

### PR DESCRIPTION
**Pull request by david-allison**
## Fixes
* Fixes #15658

## Approach
Use a toast to show the dialog instead

## How Has This Been Tested?
* ⚠️ Untested

`"$url: $message"` -> `http://127.0.0.1:46161/backend/web/deck-options.html: hello world`

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by lukstbit — APPROVED** [review]
Fine from my point of view.

A Toast "escapes" the screen where it's shown, wouldn't a SnackBar be better as it gets cancelled when the particular screen is exited?

**Comment by david-allison** [comment]
I suspect there's a bug which is causing this, I want it to be a little annoying so we get better diagnostics

**Comment by lukstbit** [comment]
> I want it to be a little annoying so we get better diagnostics

I understand, my concern is to not be so annoying that people review bomb our playstore listing.

**Comment by david-allison** [comment]
I've only seen it occur maybe 5 times, it feels like a worthwhile risk to take

**Comment by BrayanDSO** [comment]
TBH, I prefer a crash than system-wide toast spam

**Comment by david-allison** [comment]
happy to make the change, what reporting mechanism would you like?

**Comment by BrayanDSO** [comment]
> happy to make the change, what reporting mechanism would you like?

`view.url` (or some other property of `WebView`, needs testing) should at least tell us in which screen this happens so we can track what kind of `JsAlert` it emits.

the message should be useful as well

**Comment by david-allison** [comment]
`"$url: $message"` -> `http://127.0.0.1:46161/backend/web/deck-options.html: hello world`

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/PageChromeClient.kt [hunk 1]
```diff
@@ -41,8 +42,14 @@ open class PageChromeClient : WebChromeClient() {
                 positiveButton(R.string.dialog_ok) { result?.confirm() }
                 setOnCancelListener { result?.cancel() }
             }
+        } catch (e: IllegalStateException) {
+            // window count is over max!!
+            Timber.w(e, "onJsAlert: message ignored")
+            // we want to know what went wrong
+            CrashReportService.sendExceptionReport("onJsAlert:windowCount", "$url: $message")
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/pages/PageChromeClient.kt:49]: I believe that you got wrong the parameters order

https://github.com/ankidroid/Anki-Android/blob/6a41697449c76aa6bc07f63565b175dcc2ef38cc/AnkiDroid/src/main/java/com/ichi2/anki/CrashReportService.kt#L255-L258

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/pages/PageChromeClient.kt:49]: Great catch! ordering feels inconsistent there

**Review by BrayanDSO — APPROVED** [review]

**Comment by BrayanDSO** [comment]
> I've only seen it occur maybe 5 times, it feels like a worthwhile risk to take

I just went into ACRA and there are a lot of reports for that in 2.17.0

**Pull request merged**